float64 x
float64 y
float64 z
float64 xdot
float64 ydot
float64 zdot